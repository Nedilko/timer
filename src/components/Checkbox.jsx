import {useState, useEffect} from 'react'
import PropTypes from 'prop-types'

Checkbox.propTypes = {
  checked: PropTypes.bool,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
}

function Checkbox({checked = false, name, onChange}) {
  const [isChecked, setIsChecked] = useState(checked)
  const handleChange = (e) => {
    setIsChecked((oldState) => !oldState)
  }

  useEffect(() => {
    const checkbox = {}
    checkbox[name] = isChecked
    onChange(checkbox)
  }, [isChecked])

  return (
    <label htmlFor={name} className="py-2 flex flex-row items-center select-none">Use system theme
      <input onChange={handleChange} id={name} name={name} type="checkbox" value={isChecked}
             className="ml-2 w-4 h-4 text-blue-600 bg-gray-100 rounded border-gray-300 focus:ring-blue-500 focus:ring-2"/>
    </label>
  )
}

export default Checkbox