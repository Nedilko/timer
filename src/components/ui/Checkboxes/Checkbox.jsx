import PropTypes from "prop-types";

Checkbox.propTypes = {
  checked: PropTypes.bool,
  label: PropTypes.string,
  onChange: PropTypes.func.isRequired,
};

function Checkbox({ checked, label, onChange }) {
  const handleChange = (e) => {
    onChange(e.target.checked);
  };

  console.log(checked);

  return (
    <label
      htmlFor="useSystemTheme"
      className="py-2 flex flex-row items-center select-none"
    >
      {label}
      <input
        onChange={handleChange}
        id="useSystemTheme"
        type="checkbox"
        checked={checked}
        className="ml-2 w-4 h-4 text-blue-600 bg-gray-100 rounded border-gray-300 focus:ring-blue-500 focus:ring-2"
      />
    </label>
  );
}

export default Checkbox;
