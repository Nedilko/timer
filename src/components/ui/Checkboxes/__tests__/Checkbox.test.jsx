import Checkbox from "../Checkbox";
import { render, screen } from "@testing-library/react";
import useEvent from "@testing-library/user-event";

describe("Checkbox", () => {
  const clickHandler = jest.fn();

  beforeEach(() => {
    clickHandler.mockClear();
  });

  it("should be in the document", () => {
    render(<Checkbox label="test label" onChange={clickHandler} />);

    expect(screen.getByRole("checkbox")).toMatchSnapshot();
  });

  it("should have lable", () => {
    render(<Checkbox label="test label" onChange={clickHandler} />);

    expect(screen.getByLabelText("test label")).toBeInTheDocument();
  });

  it("should be renderend unchecked by default", () => {
    render(<Checkbox label="test label" onChange={clickHandler} />);

    expect(
      screen.getByRole("checkbox", { checked: false })
    ).toBeInTheDocument();
  });

  it("should be rendered checked", () => {
    render(
      <Checkbox checked={true} label="test label" onChange={clickHandler} />
    );

    expect(screen.getByRole("checkbox", { checked: true })).toBeInTheDocument();
  });

  it("should be checked on click checkbox", async () => {
    render(<Checkbox label="test label" onChange={clickHandler} />);

    screen.debug();

    expect(screen.getByRole("checkbox").checked).toEqual(false);

    await useEvent.click(screen.getByRole("checkbox"));

    screen.debug();

    expect(screen.getByRole("checkbox").checked).toEqual(true);
  });

  // it("should be checked on click label", () => {
  //   render(<Checkbox label="test label" onChange={clickHandler} />);

  //   useEvent.click(screen.getByRole("label"));

  //   expect(screen.getByRole("checkbox")).toBeInTheDocument();
  // });

  it("should hanlde click", () => {
    render(<Checkbox label="test label" onChange={clickHandler} />);

    useEvent.click(screen.getByRole("label"));

    expect(clickHandler).toHaveBeenCalledTimes(1);
  });
});
