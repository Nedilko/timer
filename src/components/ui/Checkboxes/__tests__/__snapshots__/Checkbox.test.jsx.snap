// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should be in the document 1`] = `
<input
  class="ml-2 w-4 h-4 text-blue-600 bg-gray-100 rounded border-gray-300 focus:ring-blue-500 focus:ring-2"
  id="useSystemTheme"
  role="checkbox"
  type="checkbox"
/>
`;
